### ISSUE TITLE
Overwrite resolvable dns name to local proxy [question]

### ISSUE THREAD

**Issue by kreemer** [symfony-cli/symfony-cli#284]
Currently when you start the proxy server with `symfony proxy:start` the proxy will serve a [PAC](https://developer.mozilla.org/en-US/docs/Web/HTTP/Proxy_servers_and_tunneling/Proxy_Auto-Configuration_PAC_file) file with the default url `http://127.0.0.1:7080/proxy.pac`.

The generated PAC file has following content:

```javascript
// Only proxy *.%s requests
// Configuration file in ~/.symfony5/proxy.json
function FindProxyForURL (url, host) {
	if (dnsDomainIs(host, '.%s')) {
		if (isResolvable(host)) {
			return 'DIRECT';
		}
		return 'PROXY %s';
	}
	return 'DIRECT';
}
```

[Source](https://github.com/symfony-cli/symfony-cli/blob/main/local/proxy/proxy.go#L315) 

As you can see if you specify a host which is locally resolvable, the proxy will not be set for the host. In our use case it is necessary to overwrite this behaviour for dns names, because our OIDC provider requires that the redirect URI should be a resolvable dns name.

Our current workaround is to use the symfony cli proxy mechanism and provide a PAC file ourself. We have developed a small helper script which works in our case under here: [gist](https://gist.github.com/kreemer/8c5244046f2e33b58754cac7b00a4dbd).

Is there a better workaround?

**Comment by pmishev** [comment]
This change introduced in https://github.com/symfony-cli/symfony-cli/commit/7c2f9a7c00cc716f0c73d18fc24fb567a6fd2458 is also causing problems for us.
When opening a `*.local` domain, Chrome/Chromium is stuck for 5s in `resolving host in proxy script` state on every page.

As a workaround at the moment, we either use Firefox, which doesn't seem to be affected, or set a manual HTTP(S) proxy to `127.0.0.1:7080`.

Can this change be reverted or is there a better workaround at least?

**Comment by tucksaun** [comment]
Hi folks,

I'm sorry this is causing issues for you.
I think there are two different issues here though.

Let me sum up @pmishev issue to be sure I understand:
You are using `local` as the local proxy TLD but the domain name you are using for a specific project is NOT resolvable.
Right?
If so this might be due to the fact that `local` is a bit of a special TLD.
Or that can be due to your DNS resolver.
Can you try with a different TLD and see if the same problem occurs?

Regarding @kreemer' issue, so if I got it right your OIDC requires a true DNS entry?
I'm not sure if that helps but does it prevent it to be resolved to 127.0.0.1?
The problem with this restriction is that I don't see how the previous solution would not also affect you 🤔
Because if I got it correctly, you are using an actual TLD used on the Internet, but the local proxy is only usable with local projects which means you are effectively cutting your local machine for every domain under this TLD.

Iterating on the domain list is a solution I considered when implementing #271 but I'm not really fond of it, especially as the `proxy.pac` file is hit quite a lot, but maybe this is the only viable one.

**Comment by tucksaun** [comment]
I meant what if you use an FQDN that resolves to `127.0.0.1`, something like `myapp.localtest.me`?
Sorry for the confusion if I was not clear.

Regarding your propositions, the first one will probably fix your issues but does not feel really user-friendly for you 😕
Expanding the configuration seems not really easy to maintain for everyone.

So if we want to keep the following features:
- no TLD restrictions, even one that has domains that can be resolved on the Internet
- allow bypassing the proxy for some hosts
- simple or no configuration
Then I don't really see any other option than looping over the domain list

**Comment by kreemer** [comment]
I think it would be possible to set a valid DNS record which resolves to 127.0.0.1 and bypass the proxy entirely.
I'm testing it out in the following weeks. I would close this issue now (because the question is answered). 
If the approach doesn't work, I would open a PR with the relevant changes to iterate over the domain list and write these directly in the PAC File.

@tucksaun Thanks for your help!
